docs/design_proposals/private_pip_registries.md

```diff
@@ -0,0 +1,106 @@
+# Proposal Design: Support for Private `pip` registries
+
+## Objective
+
+The aim is to support private `pip` package registries in pixi.
+
+## Motivation
+
+To deliver and deploy internal software, organizations often host packages in private registries requiring authentication. Examples of such solutions are:
+
+* Google Cloud Artifact Registry
+* AWS CodeArtifact
+* Artifactory
+* packagecloud
+* ...etc.
+
+To use such repositories with pixi, authentication should be supported.
+
+## Design considerations
+
+1. **Compatibility**. Different package servers use different ways of deriving credentials. Some use static usernames and passwords, some use temporary tokens that need to be regenerated once in a while. Ideally, `pixi` itself should not be concerned about those details.
+2. **Ease of use**. In environments where authentication credentials can be derived automatically (e.g. GCP ADC), authentication should work "natively".
+3. **Configurable**. In `pixi.toml` file, developers should be able to choose what authentication mechanism to choose for a given channel.
+4. **Separation of concerns**. Authentication implementation for different platforms can be separate from `pixi` itself.
+
+## Proposed solution
+
+### Support for authentication helpers in `pixi`
+
+Authentication helper is a program that receives an URL to the package registry as an input and outputs a JSON object containing the credentials. Authentication helpers can utilize different methods for obtaining credentials, but it's expected that they are specific to the package host implementation.
+
+Authentication helper is a runnable binary available in `PATH`. The name of the binary must start with `pixi-auth-` and followed by its unique ID.
```

Comment:
Why add this restriction now? What if another package or CLI tool already provides the needed functionality to return the username/password JSON? I would recommend to keep this more flexible, which means just allowing specifying a path to any binary / shell.

Reply:
It's a 5 minutes thing to implement something like `aws sts assume-role ... | jq '...'` which returns the desired {"username": "...", "password": "..."} JSON  compared to creating wrapper packages and shipping them to pypi/conda-forge which takes even someone experienced at least a few hours. 
My point is that almost any existing tool can be converted to the interface you are proposing with a few lines of code.